src/opencmiss_iron.f90

```diff
@@ -61522,7 +61522,9 @@ SUBROUTINE cmfe_FieldMLIO_Finalise( fieldml, err )
 
 #ifdef USEFIELDML
 
-    CALL FIELDML_IO_FINALISE( fieldml%fieldmlInfo, err, error, *999  )
+    IF( ASSOCIATED(fieldml%fieldmlInfo) ) THEN
```

Comment:
This is probably not necessary here. The coding style is that all pointers should be checked for association before they are de-referenced. As the fieldmlInfo pointer is not de-referenced it can be left for the fieldml_io_finalise routine to check the association status. However, what the problem is here is that the fieldml_info argument (first argument to fieldml_io_finalise) should be declared as a pointer in that routine. The association can then be checked in finalise and if it is associated then the finalisation can happen (the idea is that finalise can be called with a null pointer and it just does nothing). Because it is not declared as a pointer then it gets de-referenced in the subroutine call.